rudaux/submission.py

```diff
@@ -234,7 +234,7 @@ def get_latereg_overrides(extension_days, subm_set):
 
             to_remove = None
             to_create = None
-            if regdate > assignment['unlock_at']:
+            if regdate > assignment['unlock_at'] and assignment['due_at'] <= plm.from_format(config.registration_deadline, f'YYYY-MM-DD', tz=config.notify_timezone).add(days=extension_days):
```

Comment:
@trevorcampbell 

This would be the main change: only assignments that are due to the before `registration_deadline + extension_days` will be overridden. 

I had to pass the config file to the function as well, so I had to update the reference to the function in `flows.py` as well.

Reply:
You don't need the `+ extension_days` here, but otherwise it's fine!

Reply:
Are you sure? I'm thinking about an assignment deadline close to the registration deadline. 

For example, next term the registration deadline is 01/21. If an assignment is due 01/23, then the student who registered on 01/20 should still get the extension, right? However, if the student has at least the number of extension days to do the assignment, then we don't need to extend anymore. 

Does it make sense?

Reply:
Oh, yes. You're right.

The problem is that you should be using the `assignment['unlock_at']` date in your if statement. Then you can get rid of the extension days. The unlock date is what we use to determine whether someone gets an extension.

But please also open an issue for the following: not this semester (too many other things to do), but maybe in the future we should decide whether to change the extension check to (`reg_date + extension_days > due_date`).